osquery/tables/system/hash.cpp

```diff
@@ -182,6 +186,21 @@ bool FileHashCache::load(const std::string& path, MultiHashes& out) {
   return true;
 }
 
+std::string genSsdeepForFile(const std::string& path) {
+#ifndef WINDOWS
+  std::string file_ssdeep_hash(FUZZY_MAX_RESULT, '\0');
+  auto did_ssdeep_fail =
+      fuzzy_hash_filename(path.c_str(), &file_ssdeep_hash.front());
+  if (did_ssdeep_fail) {
+    LOG(WARNING) << "ssdeep failed: " << path;
+    return "-1";
+  }
+  return file_ssdeep_hash;
```

Comment:
According to https://ssdeep-project.github.io/ssdeep/doc/api/html/fuzzy_8h.html output could be less than FUZZY_MAX_RESULT. To prevent any misunderstanding and even bug it is better to trim the null-bytes-tail.
```c++
str.resize(str.find('\0', 0, str.size()));
```